### ISSUE TITLE
text_encoder was not finetuned with script examples/images/dreambooth/train_dreambooth_colossalai.py,[BUG]: 

### ISSUE THREAD

**Issue by SummerTrains** [hpcaitech/ColossalAI#3146]
### 🐛 Describe the bug

When I train dreambooth with script examples/images/dreambooth/train_dreambooth_colossalai.py, I found that the text_encode model file text_encoder/pytorch_model.bin has not changed. It is means the text encoder has not been optimized.

Then I found that only unet has been warpped with GeminiAdamOptimizer (https://github.com/hpcaitech/ColossalAI/blob/main/examples/images/dreambooth/train_dreambooth_colossalai.py#L475), but the text encoder not.

so how can I finetune both unet and text encoder?

**Comment by JThh** [comment]
Hi, currently the setting for text encoder is `requires_grad=False` as shown in [line](https://github.com/hpcaitech/ColossalAI/blob/3c01280a56353a363a031a630d7f4b6eb44b17f5/examples/images/dreambooth/train_dreambooth_colossalai.py#L463). Can you try making them trainable and later save the model states accordingly?

**Comment by SummerTrains** [comment]
@JThh thx！
I have setted text_encoder.requires_grad_(True) and text_encoder.train(), but parameter of text-encoder was not changed. I print `text_encoder.text_model.encoder.layers[3].self_attn.q_proj.weight`, it has not been changed!

**Comment by JThh** [comment]
Yes, @SummerTrains , you are right. Can you try imitating what was done to this [file](https://github.com/huggingface/diffusers/pull/883/files)?

**Comment by SummerTrains** [comment]
@JThh The most important code is ` params_to_optimize = (
        itertools.chain(unet.parameters(), text_encoder.parameters()) if args.train_text_encoder else unet.parameters()
    )`, and sand it into torch.optimizer. But it's different with GeminiAdamOptimizer.

**Comment by SummerTrains** [comment]
@JThh The most important code is  `params_to_optimize = ( itertools.chain(unet.parameters(), text_encoder.parameters()) if args.train_text_encoder else unet.parameters() )`, and sand it into torch.optimizer. But it's different with GeminiAdamOptimizer.

The key point is that ZeroOptimizer , the super class of GeminiAdamOptimizer, has a `module(ZeroDDP)` arguement, not a list of `module(ZeroDDP)`.

So how can warp the both parameters of text_encoder and unet into a GeminiAdamOptimizer

**Comment by JThh** [comment]
Would you please change [this](url) `__init__` method to possibly be:

```
    def __init__(self, models: Union[List[torch.nn.Module], torch.nn.Module] , **defaults: Any) -> None:
        if isinstance(models, list):
            parameters = itertools.chain(*[module.parameters() for module in models])
        else:
            parameters = models.parameters()
        optimizer = HybridAdam(parameters, **defaults)
        super().__init__(optimizer, model, **defaults)
```

, and import `itertools` as well as `from typing import List, Union` accordingly?

I will submit a change to this code file formally soon.

**Comment by SummerTrains** [comment]
> Would you please change [this](url) `__init__` method to possibly be:
> 
> ```
>     def __init__(self, models: Union[List[torch.nn.Module], torch.nn.Module] , **defaults: Any) -> None:
>         if isinstance(models, list):
>             parameters = itertools.chain(*[module.parameters() for module in models])
>         else:
>             parameters = models.parameters()
>         optimizer = HybridAdam(parameters, **defaults)
>         super().__init__(optimizer, model, **defaults)
> ```
> 
> , and import `itertools` as well as `from typing import List, Union` accordingly?
> 
> I will submit a change to this code file formally soon.

It's not work. Because ZeroOptimizer take `module: ZeroDDP` as input.   And I change `super().__init__(optimizer, models[0], **defaults)`, it raise some other error.

So, looking for your MR as soon as possible.

**Comment by SummerTrains** [comment]
@JThh hello, how is going?  When will you finish this problem?

**Comment by hdjsjyl** [comment]
hi @JThh , Thanks for your reply. Have you solved this issue?
